oauthenticator/globus.py

```diff
@@ -72,22 +43,58 @@ class GlobusLoginHandler(OAuthLoginHandler, GlobusMixin):
     def get(self):
         redirect_uri = self.authenticator.get_callback_url(self)
         client = self.authenticator.globus_portal_client()
+        state = self.get_state()
+        self.set_state_cookie(state)
         client.oauth2_start_flow(
             redirect_uri,
             requested_scopes=' '.join(self.authenticator.scope),
-            refresh_tokens=self.authenticator.allow_refresh_tokens
+            refresh_tokens=self.authenticator.allow_refresh_tokens,
+            state=state
```

Comment:
`state` wasn't originally checked for in the older oauthenticator. Now that it's built in, we can pass it through in the first hop and it will automatically be checked before getting to the `authenticate` method.

Reply:
What is `state`?

Reply:
From https://docs.globus.org/api/auth/developer-guide/, the `state` variable is: 

> An opaque value that is returned to the client in the authorization response. The client should use this value to prevent cross-site request forgery as described in https://tools.ietf.org/html/rfc6749#section-10.12.

It really doesn't change the end result in any way, but it makes the login process more secure.